### PR TITLE
Split controllers

### PR DESCRIPTION
**Pull request by noxone**
The class `DashboardController` has been splitted into three classes `EventSourceDescriptorController`, `EventSourceInstanceController` and `DashboardController`. Each controller has a limited scope.
The classes of the `model` package have been made unmodifiable after instantiation as far as possible (using builders or chaning the constructors).
Foreign events are handled better and optionally a authentication is required to send foreign events to the dashboard.

### CODE REVIEW

File: oversigt-ui/src/main/angular2/test.rest [hunk 1]
```diff
@@ -0,0 +1,19 @@
+POST http://localhost/events
```
Comment (lars-sh) [inline oversigt-ui/src/main/angular2/test.rest]: Did you really want to commit this?

File: core/src/main/java/com/hlag/oversigt/util/Utils.java [hunk 2]
```diff
@@ -178,6 +181,26 @@ public static String formatDuration(final Duration duration) {
...
+		return sortedSet(Arrays.asList(items));
+	}
+
+	public static Set<String> sortedSet(final Collection<String> items) {
+		final Set<String> set = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+		set.addAll(items);
+		return Collections.synchronizedSet(set);
+	}
+
+	public static <T> T getOne(@Nullable final T one, @Nullable final T two) {
```
Comment (lars-sh) [inline core/src/main/java/com/hlag/oversigt/util/Utils.java]: Try `de.larssh.utils.Optionals.getFirstValue(v -> v != null, one, two).orElseThrow()`?

File: core/src/main/java/com/hlag/oversigt/util/Utils.java [hunk 3]
```diff
@@ -178,6 +181,26 @@ public static String formatDuration(final Duration duration) {
 		return values;
 	}
 
+	public static Set<String> sortedSet(final String... items) {
+		return sortedSet(Arrays.asList(items));
+	}
+
+	public static Set<String> sortedSet(final Collection<String> items) {
```
Comment (lars-sh) [inline core/src/main/java/com/hlag/oversigt/util/Utils.java]: What about "sortedAndSynchronizedSet" or such?

File: core/src/main/java/com/hlag/oversigt/model/Dashboard.java [hunk 4]
```diff
@@ -21,20 +22,8 @@
...
-	}
-
-	private static Set<String> newSet(final Collection<String> items) {
-		final Set<String> set = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
-		set.addAll(items);
-		return Collections.synchronizedSet(set);
-	}
-
-	static final int TILE_DISTANCE = Finals.constant(6);
+	private static final int TILE_DISTANCE = /* Finals.constant(6) */ 6;
```
Comment (lars-sh) [inline core/src/main/java/com/hlag/oversigt/model/Dashboard.java]: Finals.constant should be used for non-private constants only. I'll add another checkstyle check for that.
As this code is correct, what about removing the commented code?

File: core/src/main/resources/config.json [hunk 5]
```diff
@@ -33,6 +33,13 @@
 			"uidAttribute": "",
 			"bindUser": "",
 			"bindPassword": ""
+		},
+		"foreignEvents": {
+			"enabled": true,
+			"needAuthentication": true,
+			"allowedApiKeys": [
+				"7bcc2fd0-31f1-4c35-be8c-fe0b2b484053"
```
Comment (lars-sh) [inline core/src/main/resources/config.json]: We should not deliver a default API key!

File: core/src/main/java/com/hlag/oversigt/model/Dashboard.java (hunk 4 above)
Comment (noxone) [reply core/src/main/java/com/hlag/oversigt/model/Dashboard.java]: You want the comment to be removed? Or what is your change request?

Comment (lars-sh) [reply core/src/main/java/com/hlag/oversigt/model/Dashboard.java]: Yes, simply `... TITLE_DISTANCE= 6;`

**Pull request merged**